includes/class-sensei-frontend.php

```diff
@@ -1765,13 +1771,13 @@ public function sensei_process_registration() {
 
 		// Redirect.
 		global $wp;
-		if ( wp_get_referer() ) {
-			$redirect = esc_url( wp_get_referer() );
+		if ( ! empty( $new_user_http_referer ) ) {
+			$redirect = esc_url( $new_user_http_referer );
 		} else {
 			$redirect = esc_url( home_url( $wp->request ) );
```

Comment:
I'd move the escaping down to line 1780.

Reply:
Done!